internal/build/phase_test.go

```diff
@@ -446,3 +475,62 @@ func CreateFakeLifecycleExecution(logger logging.Logger, docker client.CommonAPI
 		NoProxy:    "some-no-proxy",
 	})
 }
+
+// helper function to expose standard UNIX socket `/var/run/docker.sock` via TCP localhost:12345
+func forwardUnix2TCP(ctx context.Context, t *testing.T) {
+	wg := sync.WaitGroup{}
+
+	forwardCon := func(tcpCon net.Conn) {
+		defer wg.Done()
+
+		defer tcpCon.Close()
+		go func() {
+			<-ctx.Done()
+			tcpCon.Close()
+		}()
+
+		unixCon, err := net.Dial("unix", "/var/run/docker.sock")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer unixCon.Close()
+		go func() {
+			<-ctx.Done()
+			unixCon.Close()
+		}()
+
+		buffA := make([]byte, 4096)
+		buffB := make([]byte, 4096)
+		go io.CopyBuffer(tcpCon, unixCon, buffA)
```

Comment:
Here I am not checking errors because it's not easy to check if it really error. For instance `connection closed` is not really error but expected state at some point.